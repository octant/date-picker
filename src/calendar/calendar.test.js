/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import serializer from 'jest-glamor-react'
import sinon from 'sinon'
import Enzyme, {mount} from 'enzyme'
import Adaptor from 'enzyme-adapter-react-16'

import Calendar from './'

Enzyme.configure({adapter: new Adaptor()})
expect.addSnapshotSerializer(serializer)

describe('<Calendar /> appearance', () => {
  test('displays the same', () => {
    const component = renderer.create(
      <Calendar selected='2019-06-04' />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('<Calendar /> functionality', () => {
  test('sets an "id" property on each item', () => {
    const calendar = mount(<Calendar />)

    expect(calendar.find({id: '2018-03-22'}).length).toBe(1)
  })

  test('defaults to current month if a "selected" prop is not passed', () => {
    const date = new Date()
    const dateString = `${date.getFullYear()}-${('0' + (date.getMonth() + 1)).slice(-2)}-${('0' + date.getDate()).slice(-2)}`
    const calendar = mount(<Calendar />)

    expect(calendar.find({id: dateString}).length).toBe(1)
  })

  test('loads correct dates when passed a selected date', () => {
    const date = new Date(2017, 11, 25)
    const dateString = `${date.getFullYear()}-${('0' + (date.getMonth() + 1)).slice(-2)}-${('0' + date.getDate()).slice(-2)}`

    const calendar = mount(<Calendar selected={dateString} />)

    expect(calendar.find({id: dateString}).length).toBe(1)
  })

  test('loads the next month when next is clicked', () => {
    const calendar = mount(<Calendar selected='2018-03-22' />)

    calendar.find('button').last().simulate('click')
    calendar.find('button').last().simulate('click')

    expect(calendar.state().calendar[0].getDate()).toEqual(29)
  })

  test('loads the previous month when previous is clicked', () => {
    const calendar = mount(<Calendar selected='2018-05-22' />)

    calendar.find('button').first().simulate('click')
    calendar.find('button').first().simulate('click')

    expect(calendar.state().calendar[0].getDate()).toEqual(25)
  })

  test('the selected item is set when a Item is clicked', () => {
    const date = new Date()
    const dateString = `${date.getFullYear()}-${('0' + (date.getMonth() + 1)).slice(-2)}-${('0' + date.getDate()).slice(-2)}`
    const calendar = mount(
      <Calendar />
    )

    calendar.find({id: dateString}).first().simulate('click')

    expect(calendar.state().selected).toEqual(dateString)
  })

  test('has 42 instances of Item', () => {
    const calendar = mount(<Calendar />)

    expect(calendar.find({clickable: true}).length).toBe(42)
  })

  test('calls the clickMethod when a Item is clicked', () => {
    sinon.spy(Calendar.prototype, '_handleClick')
    const calendar = mount(
      <Calendar />
    )

    calendar.find('Item').last().simulate('click')

    expect(Calendar.prototype._handleClick.callCount).toBe(1)
    Calendar.prototype._handleClick.restore()
  })

  test('all instances of Item are clickable', () => {
    sinon.spy(Calendar.prototype, '_handleClick')
    const calendar = mount(
      <Calendar />
    )

    calendar.find('Item').forEach((item) => {
      item.simulate('click')
    })

    expect(Calendar.prototype._handleClick.callCount).toBe(42)
    Calendar.prototype._handleClick.restore()
  })
})
