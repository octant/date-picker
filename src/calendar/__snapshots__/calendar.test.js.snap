// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> appearance displays the same 1`] = `
.css-0,
[data-css-0] {
  position: absolute;
  background-color: cyan;
  padding: 0.5em;
}

.css-1,
[data-css-1] {
  position: relative;
  overflow: hidden;
  padding: 0.5em;
  background-color: silver;
}

.css-2,
[data-css-2] {
  position: relative;
  width: 17.5em;
}

.css-3,
[data-css-3] {
  position: relative;
  height: 2.5em;
  width: 2.5em;
  float: left;
  margin: 0em;
  line-height: 2.5em;
  text-align: center;
  background-color: orangered;
}

.css-4,
[data-css-4] {
  position: relative;
  height: 2.5em;
  width: 2.5em;
  float: left;
  margin: 0em;
  line-height: 2.5em;
  text-align: center;
  background-color: orangered;
}

.css-4:hover,
[data-css-4]:hover {
  cursor: pointer;
  color: white;
  background-color: darkblue;
}

<div
  className="css-0"
>
  <h1>
    2019-06-04
  </h1>
  <div
    className="css-1"
  >
    <div>
      <button
        onClick={[Function]}
      >
         
        &lt;
         
      </button>
       
      <button
        onClick={[Function]}
      >
         
        &gt;
         
      </button>
    </div>
    <div
      className="css-2"
    >
      <div
        className="css-3"
        onClick={[Function]}
      >
        Su
      </div>
      <div
        className="css-3"
        onClick={[Function]}
      >
        Mo
      </div>
      <div
        className="css-3"
        onClick={[Function]}
      >
        Tu
      </div>
      <div
        className="css-3"
        onClick={[Function]}
      >
        We
      </div>
      <div
        className="css-3"
        onClick={[Function]}
      >
        Th
      </div>
      <div
        className="css-3"
        onClick={[Function]}
      >
        Fr
      </div>
      <div
        className="css-3"
        onClick={[Function]}
      >
        Sa
      </div>
    </div>
    <div
      className="css-2"
    >
      <div
        className="css-4"
        onClick={[Function]}
      >
        26
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        27
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        28
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        29
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        30
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        31
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        2
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        3
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        4
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        5
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        6
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        7
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        8
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        9
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        10
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        11
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        12
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        13
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        14
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        15
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        16
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        17
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        18
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        19
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        20
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        21
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        22
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        23
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        24
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        25
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        26
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        27
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        28
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        29
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        30
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        2
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        3
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        4
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        5
      </div>
      <div
        className="css-4"
        onClick={[Function]}
      >
        6
      </div>
    </div>
  </div>
</div>
`;
